controllers/topics/checkName.js

```diff
@@ -0,0 +1,37 @@
+const {Topics} = require('../../databases/models');
+
+module.exports = async (req, res) => {
+  try {
+    let {name} = req.body;
+    name = name.toLowerCase();
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    let response = {
+      success: false,
+      message: 'A community with this name already exists',
+      available: false
+    };
+
+    if (topics) {
+      response = {
+        success: true,
+        message: 'Community name available',
+        available: true
+      };
+    }
+
+    return res.status(200).json(response);
+  } catch (error) {
+    return res.json({
+      code: error.statusCode,
+      status: 'fail',
+      message: error.message,
+      data: 'null'
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling could be improved here. Instead of returning a generic error message, consider sending a more specific HTTP status code depending on the type of error. For instance, you can return a 500 status code for server errors. Also, it's better to log the error for debugging purposes.

```diff
-    return res.json({
-      code: error.statusCode,
-      status: 'fail',
-      message: error.message,
-      data: 'null'
-    });

+    console.error(error);
+    return res.status(500).json({
+      status: 'error',
+      message: 'An error occurred while checking community name availability'
+    });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->